force-app/main/default/classes/SOQL.cls

```diff
@@ -271,11 +403,18 @@ public with sharing class SOQL {
      */
     public static List<Contact> getAccountsWithContactsByState(Id accountId) {
         // Query the Account
+        Account acc = [SELECT BillingState FROM Account WHERE Id = :accountId LIMIT 1];
 
         // Store the Account's BillingState
+        String billingState = acc.BillingState;
 
         // Query the Contacts
-        return null; // Replace null with the variable you used to store the result
+        List<Contact> contacts = [SELECT Id, FirstName, LastName, MailingState FROM Contact
+                                WHERE MailingState = :billingState];
+
```

Comment:
Can you find a way to do this with just 1 query? HINT: you can use a subquery in the criteria